rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,233 @@
+
+The actual grouping of the files is to be performed by analysis via a search in the metadata of the bundles submitted following well-defined modality-specific set of rules. For this reason, the metadata must satisfy the following metadata requirements (MR):
+
+Furthermore, if part of the data submitted in a DAPS has already been processed the Analysis infrastructure will only re-process the parts that have not been modified.
+
+1. The data modality of each data file included in the DAPS must be clearly identifiable (MR1)
+2. Data that need to be processed together must be clearly identifiable by a modality-specific set of metadata variables that will group the files according to processing requirements. (MR2)
+3. Among the data that needs to be processed together, the metadata must provide sufficient information to correctly order, pair or otherwise establish correspondence and type of the input files. (MR3)
+
+## Proposed Signalling Implementation
+The DAPS proposal above only sets specific requirements for the signal to initiate data analysis and minimum requirements for the associated metadata. We hereby provide a specific recommendation for implementation via bundle notifications.
+
+We recommend that the data store (DSS) is used to store DAPS as a new bundle type of **“DAPS Manifest”**, and that the current search-based notification system is adjusted so that instead of listening for new primary bundles, analysis listens for new DAPS bundles that are exclusively used for the initiation of pipelines. This implementation has two major advantages: (1) the DAPS are permanently saved at the main storage site for the HCA project (DSS) so the analyses can be reproducibly triggered and (2) The existing notification mechanisms are reused, avoiding the need for implementation of a new out-of-band notification system.
+
+We propose that as an initial implementation approach DAPS are used at set data aggregation levels (e.g. sample, project) but critically the implementation must be flexible to arbitrary data aggregation levels to accommodate future project needs. We propose that triggering is initially done manually by data wranglers and/or ingest, but we envisage transition to automatic triggering in the future.
+
+## Example: Analysis Implementation for a DAPS
+The following pseudocode presents a proposed implementation of the processing of an incoming DAPS by analysis:
+```
+Partition the datasets by data modality (MR1)
+For each data modality:
+    Use the modality-specific criteria to identify pipeline invocations (MR2)
+    For each invocation:
+        order the data as required (MR3) and launch pipeline invocation
+```
+
+Note that the above approach does not require all the data in a submitted DAPS to be of the same modality to be processed correctly. For example a DAPS at the project level can contain imaging and 10X datasets and these will be correctly processed in an independent manner. 
```

Comment:
If being present together in the same project does not guarantee the need for uniform and grouped processing why create a project-level DAPS? what function does it serve?

Reply:
DAPS is a signaling mechanism that a group of bundles is available.  Right now it is impossible to know when data is complete beyond a bundle.  There is no way for ingest, or any other user, to know if a project or some subset is complete.

Reply:
I suspect with some lifecycle diagrams and pointers to how and when the DAPS bundle would be created it will be easier to see how this is different from a bundle which contains everything from a project